libraries/dateformatter/impl/src/main/kotlin/io/element/android/libraries/dateformatter/impl/di/DateFormatterModule.kt

```diff
@@ -25,5 +26,5 @@ object DateFormatterModule {
     fun providesLocale(): Locale = Locale.getDefault()
 
     @Provides
-    fun providesTimezone(): TimeZone = TimeZone.currentSystemDefault()
+    fun providesTimezone(): TimezoneProvider = TimezoneProvider { TimeZone.currentSystemDefault() }
```

Comment:
I am wondering if requesting `TimeZone.currentSystemDefault()` for all Events could come at a cost. We could maybe have a flow of  `TimeZone` that emit new item when the timezone change?

Reply:
There might be some tiny overhead from instantiating the value, but I don't think it's noticeable at all. The underlying code is cloning an existing value AFAICT, then just wrapping it in a new object. Reusing the observer for this feels like an overkill, to be honest.